airflow/providers/amazon/aws/hooks/redshift_cluster.py

```diff
@@ -153,3 +153,18 @@ def create_cluster_snapshot(
             ManualSnapshotRetentionPeriod=retention_period,
         )
         return response['Snapshot'] if response['Snapshot'] else None
+
+    def get_cluster_snapshot_status(self, snapshot_identifier: str, cluster_identifier: str):
+        try:
+            response = self.get_conn().describe_cluster_snapshots(
+                ClusterIdentifier=cluster_identifier,
+                SnapshotIdentifier=snapshot_identifier,
+            )
+            snapshot = response.get("Snapshots")[0]
+            snapshot_status: str = snapshot.get("Status")
+            return snapshot_status
+        except ClientError as exception:
+            if exception.response.get("Error", {}).get("Code", "") == "ClusterSnapshotNotFound":
+                return "cluster_snapshot_not_found"
```

Comment:
I dont really like this magic string. I would return `None` instead. Could you also add docstring to make it clear for users

Reply:
I think returning None would be cleaner here but just we have a similar method `cluster_status` which returns `cluster_not_found`  https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/redshift_cluster.py#L89
- Add docstring and now returning None

Reply:
That's a good point but I rather break the inconsistency than keep doing the "bad" thing. What we can do is duplicate `cluster_status` and name it `get_cluster_status`. `get_cluster_status` would return `None` instead of `cluster_not_found`. Use `get_cluster_status` instead of `cluster_status` and mark the latter as deprecated

Reply:
But that's optional. I am good with your current changes